src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1695,6 +1807,13 @@ void MergeTreeData::renameTempPartAndReplace(
         addPartContributionToColumnSizes(part);
     }
 
+    auto part_in_memory = asInMemoryPart(part);
+    if (part_in_memory && getSettings()->in_memory_parts_enable_wal)
+    {
+        auto wal = getWriteAheadLog();
+        wal->addPart(part_in_memory->block, part_in_memory->name);
```

Comment:
We don't create any disk reservations for this. So our WAL is not accounted in our disk space. Maybe we can use a reservation from the part?

Reply:
Now `write_ahead_log_max_bytes` are reserved